src/backend/distributed/commands/role.c

```diff
@@ -276,3 +508,136 @@ makeDefElemInt(char *name, int value)
 {
 	return makeDefElem(name, (Node *) makeInteger(value), -1);
 }
+
+
+/*
+ * GetDatabaseNameFromDbRoleSetting performs a lookup, and finds the database name
+ * associated with a Role Setting
+ */
+char *
+GetDatabaseNameFromDbRoleSetting(HeapTuple tuple, TupleDesc DbRoleSettingDescription)
+{
+	bool isnull;
+
+	Datum setdatabase = heap_getattr(tuple, Anum_pg_db_role_setting_setdatabase,
+									 DbRoleSettingDescription, &isnull);
+
+	if (isnull)
+	{
+		return NULL;
+	}
+
+	Oid databaseId = DatumGetObjectId(setdatabase);
+	char *databaseName = get_database_name(databaseId);
+
+	return databaseName;
+}
+
+
+/*
+ * GetDatabaseNameFromDbRoleSetting performs a lookup, and finds the role name
+ * associated with a Role Setting
+ */
+char *
+GetRoleNameFromDbRoleSetting(HeapTuple tuple, TupleDesc DbRoleSettingDescription)
+{
+	bool isnull;
+
+	Datum setrole = heap_getattr(tuple, Anum_pg_db_role_setting_setrole,
+								 DbRoleSettingDescription, &isnull);
+
+	if (isnull)
+	{
+		return NULL;
+	}
+
+	Oid roleId = DatumGetObjectId(setrole);
+	char *roleName = GetUserNameFromId(roleId, true);
+
+	return roleName;
+}
+
+
+/*
+ * MakeSetStatementArgsList parses a configuraton value and creates an A_Const
+ * with an appropriate type.
+ *
+ * The allowed A_Const types are Integer, Float, and String.
+ */
+List *
+MakeSetStatementArgsList(char *configurationValue)
+{
+	char *endPointer;
+
+	/* try to parse the configuration value as an integer */
+	long integerCandidate = strtol(configurationValue, &endPointer, 10);
```

Comment:
Should use SafeStringToInt64

Reply:
I used the `SafeStringTo*` functions here and introduced a new `SafeStringToFloat()`